### PR TITLE
Add Actor Timer/Reminder option builders

### PR DESCRIPTION
**Pull request by addjuarez**
Signed-off-by: addjuarez <addiajuarez@gmail.com>

# Description

_Please explain the changes you've made_

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #681 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Review by tanvigour — APPROVED** [review]

File: sdk-actors/src/main/java/io/dapr/actors/runtime/ReminderOptions.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  }
+    
+  private ReminderOptions(ReminderOptionsBuilder builder) {
+    this.name = builder.name;
+    this.dueTime = builder.dueTime;
+    this.period = builder.period;
+  }
+    
+  //Builder Class
+  public static class ReminderOptionsBuilder<T> {
```
Comment (mukundansundar) [inline sdk-actors/src/main/java/io/dapr/actors/runtime/ReminderOptions.java:47]: I think we specifically moved away from boiler pattern in java sdk for model classes https://github.com/dapr/java-sdk/issues/587. 
Can this be changed to reflect that?

File: sdk-actors/src/main/java/io/dapr/actors/runtime/TimerOptions.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  public Duration getDueTime() {
+    return dueTime;
+  }
+
+  public Duration getPeriod() {
+    return period;
+  }
+    
+  private TimerOptions(TimerOptionsBuilder builder) {
```
Comment (mukundansundar) [inline sdk-actors/src/main/java/io/dapr/actors/runtime/TimerOptions.java:44]: same here .

**Review by mukundansundar — CHANGES_REQUESTED** [review]
please remove builder pattern for model classes see https://github.com/dapr/java-sdk/issues/587

File: sdk-actors/src/main/java/io/dapr/actors/runtime/TimerOptions.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    this.name = builder.name;
+    this.callback = builder.callback;
+    this.dueTime = builder.dueTime;
+    this.period = builder.period;
+  }
+    
+  //Builder Class
+  public static class TimerOptionsBuilder<T> {
```
Comment (mukundansundar) [inline sdk-actors/src/main/java/io/dapr/actors/runtime/TimerOptions.java:52]: same here

**Comment by addjuarez** [comment]
Can you clarify what changes are needed? I removed the builder and tried using https://azure.github.io/azure-sdk/java_introduction.html#java-models-builder

**Review by artursouza — CHANGES_REQUESTED** [review]
We need unit tests and support for ISO duration.

**Pull request closed without merging**